packages/@vuepress/core/lib/client/root-mixins/updateMeta.js

```diff
@@ -1,38 +1,53 @@
+import unionBy from 'lodash/unionBy'
+
 export default {
+  // created will be called on both client and ssr
   created () {
     if (this.$ssrContext) {
+      const siteMetaTags = this.$site.headTags
+        .filter(item => item[0] === 'meta')
+        .map(item => item[1])
+
+      const meta = this.getMergedMetaTags(siteMetaTags)
+
       this.$ssrContext.title = this.$title
       this.$ssrContext.lang = this.$lang
-      this.$ssrContext.description = this.$page.description || this.$description
+      this.$ssrContext.pageMeta = renderPageMeta(meta)
     }
   },
-
+  // Other life cycles will only be called at client
   mounted () {
+    // init currentMetaTags from DOM
+    this.currentMetaTags = [...document.querySelectorAll('meta')]
+
+    // indirectly init siteMetaTags from DOM
+    this.siteMetaTags = this.currentMetaTags.map(element => {
+      const siteMeta = {}
+      for (const attribute of element.attributes) {
+        siteMeta[attribute.name] = attribute.value
+      }
+      return siteMeta
+    })
```

Comment:
```suggestion
    this.siteMetaTags = this.currentMetaTags.map(metaTag =>
      metaTag.attributes.reduce((acc, { name, value }) => ({
        ...acc,
        [name]: value
      }), {}))
```

Reply:
Sorry, I just realize that I'm init siteMetaTags with entry-page meta tags instead of site meta tags. Fixed in https://github.com/vuejs/vuepress/pull/2164/commits/a853352c166623500b5a89a8c3bb8fa58e8416e4